src/pybind/mgr/dashboard/frontend/src/app/shared/datatable/table-pagination/table-pagination.component.ts

```diff
@@ -0,0 +1,110 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+
+@Component({
+  selector: 'cd-table-pagination',
+  templateUrl: './table-pagination.component.html',
+  styleUrls: ['./table-pagination.component.scss']
+})
+export class TablePaginationComponent {
+  private _size = 0;
+  private _count = 0;
+  private _page = 1;
+  pages: any;
+
+  @Input()
+  set size(value: number) {
+    this._size = value;
+    this.pages = this.calcPages();
+  }
+
+  get size(): number {
+    return this._size;
+  }
+
+  @Input()
+  set page(value: number) {
+    this._page = value;
+  }
+
+  get page(): number {
+    return this._page;
+  }
+
+  @Input()
+  set count(value: number) {
+    this._count = value;
+  }
+
+  get count(): number {
+    return this._count;
+  }
+
+  get totalPages(): number {
+    const count = this.size < 1 ? 1 : Math.ceil(this._count / this._size);
+    return Math.max(count || 0, 1);
+  }
+
+  @Output() pageChange: EventEmitter<any> = new EventEmitter();
+
+  canPrevious(): boolean {
+    return this._page > 1;
+  }
+
+  canNext(): boolean {
+    return this._page < this.totalPages;
+  }
+
+  prevPage(): void {
+    this.selectPage(this._page - 1);
+  }
+
+  nextPage(): void {
+    this.selectPage(this._page + 1);
+  }
+
+  selectPage(page: number): void {
+    if (page > 0 && page <= this.totalPages && page !== this.page) {
+      this._page = page;
+      this.pageChange.emit({
+        page
+      });
+    } else if (page > 0 && page >= this.totalPages) {
+      this._page = this.totalPages;
+      this.pageChange.emit({
+        page: this.totalPages
+      });
+    }
+  }
+
+  calcPages(page?: number): any[] {
+    const pages = [];
+    let startPage = 1;
+    let endPage = this.totalPages;
+    const maxSize = 5;
+    const isMaxSized = maxSize < this.totalPages;
+
```

Comment:
This is a nitpick from my side but you don't need a list of pages to know in which page you are. `offset = page_num * limit_per_page` where `page_num >= 0 && page_num < max_pages`

Reply:
Thanks @pereman2, Is this the offset from the main table component ?